project-addons/account_payment_term_and_mode/models/payment_return.py

```diff
@@ -0,0 +1,67 @@
+from openerp import models, fields, api, _
+from openerp.exceptions import Warning as UserError
+
+
+class PaymentReturn(models.Model):
+
+    _inherit = 'payment.return'
+
+    # Override button function
+    @api.multi
+    def action_confirm(self):
```

Comment:
Si sólo quieres marcar como sin seguimiento las devoluciones de cobro había que intentara hacerlo sin pisar el método completo ya que se rompe la herencia. Una solución sería, después e llamar al super del método y de que se ejecute, recorrerse los apuntes del asiento de la devolución, que lo guarda en el campo "move_id" de esta y marcarlos como sin seguimiento en los que proceda.

Reply:
Totalmente cierto, al tener el código  dentro de un for pense que no podía sacarlo fuera.
He intentado hacerlo así:     
@api.multi
    def action_confirm(self):
        res = super(PaymentReturn, self).action_confirm()

        # Mark negative line as no-followup
        for line_id in self.move_id.line_id:
            if line_id.debit - line_id.credit < 0:
                line_id.write({'blocked': True})

        return res

Sin embargo, no actualiza el valor en el formulario de odoo, aparece el campo 'sin seguimiento' desmarcado, aunque en la bbdd si que está a true.

Reply:
¿y al volver a entrar? es raro que no lo haga

Reply:
Ahora si, estaba updateando una fila que no se mostraba en la tabla. Veo un poco enrevesado la forma en lo que lo he hecho, no se como lo verás.